mlflow/projects/__init__.py

```diff
@@ -422,13 +422,17 @@ def _create_run(uri, experiment_id, work_dir, entry_point):
         parent_run_id = existing_run.info.run_uuid
     else:
         parent_run_id = None
-    active_run = tracking.MlflowClient().create_run(
+    client = tracking.MlflowClient()
+    active_run = client.create_run(
         experiment_id=experiment_id,
         source_name=source_name,
         source_version=_get_git_commit(work_dir),
         entry_point_name=entry_point,
         source_type=SourceType.PROJECT,
         parent_run_id=parent_run_id)
+    if mode == "local" and databricks_utils.is_in_databricks_notebook():
+        for tag_key, tag_val in databricks_utils.get_databricks_tags():
+            client.set_tag(active_run.info.run_uuid, tag_key, tag_val)
```

Comment:
Is this code replicated somewhere else already to set tags when you are not using projects.run? If so we could factor it into a single function.

Reply:
Actually, one thing I'm realizing when looking at the fluent.py code / UI code - a notebook link will only show up if we set the source type of the run to "NOTEBOOK", and we otherwise try to show a link to the commit of the project's GitHub repo ([see UI logic here](https://github.com/mlflow/mlflow/blob/master/mlflow/server/js/src/utils/Utils.js#L135)).

There's actually a question of whether it makes more sense to link to the notebook or project's GitHub repo (probably makes the most sense to link to the project repo), in which case setting notebook tags isn't as useful.

Reply:
That's a good point. It sounds like the UI could look for the tags but would want to do something different in this case (i.e. say that this run was called in this notebook).